src/client/tensorBoard/tensorBoardSession.ts

```
@@ -0,0 +1,258 @@
+        if (await this.installer.isInstalled(Product.tensorboard)) {
+            return;
+        }
+        const interpreter = await this.interpreterService.getActiveInterpreter();
+        const tokenSource = new CancellationTokenSource();
+        const installerToken = tokenSource.token;
+        const cancellationPromise = createPromiseFromCancellation({
+            cancelAction: 'resolve',
+            defaultValue: InstallerResponse.Ignore,
+            token: installerToken
+        });
+        const response = await Promise.race([
+            this.installer.promptToInstall(Product.tensorboard, interpreter, installerToken),
+            cancellationPromise
+        ]);
+        if (response !== InstallerResponse.Installed) {
+            throw new Error(TensorBoard.tensorBoardInstallRequired());
+        }
+    }
+
+    // Display an input box asking the user for an absolute or relative log directory
+    // to tfevent files. Default this to the directory that the active text editor is in,
+    // if any, then the folder that is open in the editor, if any.
+    private async askUserForLogDir(): Promise<string> {
+        const options = {
+            prompt: TensorBoard.logDirectoryPrompt(),
+            value: this.autopopulateLogDirectoryPath(),
+            placeHolder: TensorBoard.logDirectoryPlaceholder(),
+            ignoreFocusOut: true,
+            validateInput: (value: string) => {
+                return value.trim().length > 0 ? undefined : TensorBoard.invalidLogDirectory();
+            }
+        };
+        const logDir = await window.showInputBox(options);
+
+        // Even though we validateInput above, the result of showInputBox may still be
+        // null if the user hit `esc`. The user may also have provided a log directory
+        // that does not exist. Validate it and fail fast here.
+        if (!logDir || !(await this.isValidLogDirectory(logDir))) {
+            throw new Error(TensorBoard.invalidLogDirectory());
```

Again, if user chooses not to provide a log directlry, not sure we should throw an error and display an `Error` message & log errors. 
Not sure about the UX.

To me this feels like a user is asked to enter some information to configure a tool and they chose not to configure it and we're throwing errors.
When infact we should handle that workflow more gracefully.

---

Optionally you can create a new Error class, and then when you handle this error you can check the type of the error and ignore logging that error & ignore displaying error messages.